### PR TITLE
fix #269952: allowed new templates to be shown in new wizard w/o restart

### PR DESCRIPTION
**Pull request by finleylau**
readded the template page when opening a NewFileWizard so that it reiterated over the template file folder each time to generate any new templates being saved

### CODE REVIEW

File: mscore/file.cpp [hunk 1]
```diff
@@ -480,8 +480,11 @@ MasterScore* MuseScore::getNewFile()
       {
       if (!newWizard)
             newWizard = new NewWizard(this);
-      else
+      else {
+            newWizard->removePage(newWizard->Page::Template);
```
Comment (jeetee) [inline mscore/file.cpp]: What happens with the previously created NewWizardPage4 object? Does it still need to be deleted?

File: mscore/newwizard.cpp [hunk 2]
```diff
@@ -357,6 +327,53 @@ QString NewWizardPage4::templatePath() const
...
+      search->setSizePolicy(QSizePolicy::Preferred, QSizePolicy::Fixed);
+      searchLayout->addSpacerItem(new QSpacerItem(0, 0, QSizePolicy::MinimumExpanding, QSizePolicy::Maximum));
+      searchLayout->addWidget(search);
+
+      newLayout->addLayout(searchLayout);
+      newLayout->addWidget(templateFileBrowser);
+
+      setLayout(newLayout);
+
+      connect(templateFileBrowser, SIGNAL(scoreSelected(const QString&)), SLOT(templateChanged(const QString&)));
```
Comment (anatoly-os) [inline mscore/newwizard.cpp]: Are you sure that connects are required on each `setTemplateLayout()` call? Previously, it was set once in ctor. Now, it is set each update.

Comment (finleylau) [reply mscore/newwizard.cpp]: I'm not sure as I'm not too familiar with QT's GUI classes.  For one, the connects require variables that are updated within the update function (search).  Also, I read [here](http://doc.qt.io/qt-5/qobject.html#dtor.QObject) that for a QObject's deconstructor:

> All signals to and from the object are automatically disconnected, and any pending posted events for the object are removed from the event queue. 

which is why I reconnected every time, although I may have assumed wrong.

Comment (finleylau) [reply mscore/newwizard.cpp]: I've tested not connecting on each update and the layouts don't update correctly when connecting only once in the constructor @anatoly-os

**Comment by finleylau** [comment]
@jeetee Thanks for the suggestion - that is a much better solution and worked when I tested it.  I've updated the PR

**Comment by finleylau** [comment]
has someone had a chance to review this since the last update?

File: mscore/newwizard.cpp [hunk 3]
```diff
@@ -357,6 +357,24 @@ QString NewWizardPage4::templatePath() const
...
 
+//---------------------------------------------------------
+//   setTemplateLayout
+//---------------------------------------------------------
+
+void NewWizardPage4::setTemplateLayout()
+      {
+      QDir dir(mscoreGlobalShare + "/templates");
+      QFileInfoList fil = dir.entryInfoList(QDir::NoDotAndDotDot | QDir::Readable | QDir::Dirs | QDir::Files, QDir::Name);
+      if(fil.isEmpty()){
```
Comment (Jojo-Schmitz) [inline mscore/newwizard.cpp]: space between `if`and `(`, `{...}` not needed

**Comment by Jojo-Schmitz** [comment]
Tested #3771 in 2.3, works fine, so this here seems good to merge (except for those minor stylistic quirks)

**Comment by Jojo-Schmitz** [comment]
Much better (and simpler) approach for 2.3 in #3773, same could and should be used in master, once #3741 is merged, making this PR here obsolete

**Comment by anatoly-os** [comment]
https://github.com/musescore/MuseScore/pull/3773 requires additional work to be done since there are changes in master related to `updateNewWizard` method and code around.
Anyway, this is up to separate PR.

**Pull request closed without merging**